### PR TITLE
Send users a confirmation link instead of a random password upon sign up - v2

### PR DESCRIPTION
**Pull request by nickbarnwell**
Same as the last one, except based on master. 
A few more changes coming:
- [ ] refactoring host promotion to use GetOrCreateUser
- [ ] New hosts w/o account can set their own password
- [ ] Integration Test for Confirmation Links (maybe…)

### CODE REVIEW

**Comment by nickbarnwell** [comment]
nm don't review yet – have commits on another machine

File: app/views/devise/confirmations/new.html.erb [hunk 1]
```diff
@@ -1,12 +1,15 @@
-<h2>Resend confirmation instructions</h2>
+<h2>Account Activation</h2>
 
-<%= form_for(resource, as: resource_name, url: confirmation_path(resource_name), html: { method: :post }) do |f| %>
+<%= form_for resource, :as => resource_name, :url => update_user_confirmation_path, :html => {:method => 'patch'}, :id => 'activation-form' do |f| %>
```
Comment (mmahalwy) [inline app/views/devise/confirmations/new.html.erb]: Oh no, rockets hurt me :(

Comment (mehulkar) [inline app/views/devise/confirmations/new.html.erb]: me2

**Comment by nickbarnwell** [comment]
Srsly getting this in this weekend. No, guys, really. I swear

**Comment by ankitshah811** [comment]
@nickbarnwell bump

**Comment by nickbarnwell** [comment]
Realised this needed a bunch of refactoring to simplify logic around when a user's PW gets sent to them etc., so slight delay.

**Comment by nickbarnwell** [comment]
grr fucked up the rebase will cherry-pick and merge later

**Comment by mehulkar** [comment]
@nickbarnwell is still a thing we need now that users put in passwords with they sign up?

**Comment by mehulkar** [comment]
closing due to inactivity. pushed this branch up to my fork just in case it has something absolutely essential that we can't recreate.

**Pull request closed without merging**